### ISSUE TITLE
About contrastive  loss 

### ISSUE THREAD

**Issue by xuclear** [samgregoost/atmg#3]
Hi! 
  I have a confusion about contrastive learning loss in hyperbolic space. I noticed that in your implementation,  “-” is applied to the computation of "text_logits", but not to "image_logits."

> Compute logits for hyperbolic angle based contrastive loss.
> text_logits = -L.pairwise_oxy_angle(text_feats, all_image_feats, _curv)
> image_logits = L.pairwise_oxy_angle(image_feats, all_text_feats, _curv)

 From my understanding, this contrastive loss should be symmetric, and thus both"text_logits" and "image_logits"  should either include or exclude the sign consistently.

**Comment by seunghub** [comment]
Hello,

I noticed some inconsistencies between the code and the paper, and I would appreciate clarification:

1. In the code, angles between images and text are calculated mutually, while the paper suggests that only the angle of the image with respect to the text matters.
2. Additionally, the centroid loss seems not to be utilized in the code but is utilized in the paper.

These discrepancies make it unclear what the correct implementation should be.
Thank you in advance for your guidance and clarification!
